File: SWIFT/Cap08/c08ex04.swift
// c08ex04.swift

import Foundation

var Nomes: [String] = Array(repeating: "", count: 5)
var X: String = ""

print("CLASSIFICACAO DE NOMES (DECRESCENTE)\n")

for i in 0...4 {
    print("Entre o \(i+1)o. nome: ", terminator: "")
    if let input = readLine() {
        Nomes[i] = input.trimmingCharacters(in: .newlines)
    }
}

for i in 0...3 {
    for j in i+1...4 {
        if (Nomes[i] < Nomes[j]) {
            X = Nomes[i]
            Nomes[i] = Nomes[j]
            Nomes[j] = X
        }
    }
}

print("\n")
for i in 0...4 {
    print("\(i+1)o. nome: \(Nomes[i])")
}

print()
print("Tecle <Enter> para encerrar... ", terminator: "")
_ = readLine()

File: CPP/Cap08/c08ex04.cpp
// c08ex04.cpp

#include <iostream>
#include <string>
using namespace std;

int main(void) {
  string Nomes[5];
  string X;
  int I, J;

  cout << "CLASSIFICACAO DE NOMES (DECRESCENTE)\n\n";

  for (I = 0; I <= 4; ++I) {
    cout << "Entre o " << I + 1 << "o. nome: ";
    getline(cin, Nomes[I]);
  }

  for (I = 0; I <= 3; ++I) {
    for (J = I + 1; J <= 4; ++J) {
      if (Nomes[I] > Nomes[J]) {
        X = Nomes[I];
        Nomes[I] = Nomes[J];
        Nomes[J] = X;
      }
    }
  }

  cout << endl;
  for (I = 0; I <= 4; ++I)
    cout << I + 1 << "o. nome: " << Nomes[I] << endl;

  cout << "\nTecle <Enter> para encerrar... ";
  cin.get();

  return 0;
}
